generic_tracers/generic_COBALT.F90

```diff
@@ -496,456 +508,507 @@ subroutine user_add_params
+      call get_param(param_file, "generic_COBALT", "case2_salt",     cobalt%case2_salt,  &
+                     "Salinity threshold to identify Case 2 water when using a modified opacity.", units="PSU", default=30.0 )                    ! PSU
+      call get_param(param_file, "generic_COBALT", "case2_opac_add", cobalt%case2_opac_add, &     ! m-1
+                     "Additional opactity added in Case 2 waters when using a modified opacity.", units="m-1", default=0.05 )
     else
       cobalt%case2_depth = 0.0                                                           ! m
       cobalt%case2_salt = 0.0                                                            ! PSU
       cobalt%case2_opac_add = 0.0                                                        ! m-1
     endif
-    call g_tracer_add_param('min_daylength', cobalt%min_daylength, 6.0 )                 ! hours
-    call g_tracer_add_param('refuge_conc', cobalt%refuge_conc, 1.0e-10)                  ! moles N kg-1
+    call get_param(param_file, "generic_COBALT", "min_daylength",       cobalt%min_daylength,           "min_daylength",       units="hours",       default= 6.0 )                 ! hours
+    call get_param(param_file, "generic_COBALT", "refuge_conc",         cobalt%refuge_conc,             "refuge_conc",         units="mol kg-1",    default= 1.0e-10)                  ! moles N kg-1
     !
     !-----------------------------------------------------------------------
     ! Nitrogen fixation inhibition parameters
     !-----------------------------------------------------------------------
     !
-    call g_tracer_add_param('o2_inhib_Di_pow', cobalt%o2_inhib_Di_pow, 4.0)                 ! mol O2-1 m3
-    call g_tracer_add_param('o2_inhib_Di_sat', cobalt%o2_inhib_Di_sat, 3.0e-4)              ! mol O2 kg-1
+    call get_param(param_file, "generic_COBALT", "o2_inhib_Di_pow", cobalt%o2_inhib_Di_pow, "o2_inhib_Di_pow", units="mol O2-1 m3",   default=4.0)                 ! mol O2-1 m3
+    call get_param(param_file, "generic_COBALT", "o2_inhib_Di_sat", cobalt%o2_inhib_Di_sat, "o2_inhib_Di_sat", units="mol kg-1",      default=3.0e-4)              ! mol O2 kg-1
     !
     !-----------------------------------------------------------------------
     ! Other stoichiometry
     !-----------------------------------------------------------------------
     !
-    call g_tracer_add_param('p_2_n_static', cobalt%p_2_n_static, .false. )
-    call g_tracer_add_param('c_2_n', cobalt%c_2_n, 106.0 / 16.0)
-    call g_tracer_add_param('alk_2_n_denit', cobalt%alk_2_n_denit, 552.0/472.0)             ! eq. alk mol NO3-1
-    call g_tracer_add_param('alk_2_nh4_amx', cobalt%alk_2_nh4_amx, 3732.0/8030.0)           ! eq. alk mol NH4-1
-    call g_tracer_add_param('p_2_n_static_Di', phyto(DIAZO)%p_2_n_static,1.0/40.0 )         ! mol P mol N-1
-    call g_tracer_add_param('p_2_n_static_Lg', phyto(LARGE)%p_2_n_static,1.0/14.0 )         ! mol P mol N-1
```

Comment:
@charliestock @amoebaliz @andrew-c-ross 
Can these 1.0/X parameters be changed to X and then we can do the 1.0/X calculation on the next line? 
Some of the resulting defaults cannot be exactly represented in the parameter file.

Reply:
@theresa-morrison, I would suggest that we probably do not want users to change these `class stoichiometry` parameters at this moment. Perhaps we could hide them from `COBALT_parameter_doc.all` if there is an option in the `get_param` call to do so.

Reply:
I think we a reasonable level of accuracy is fine here (i.e., we could convert to a decimal value if needed).  That said, I agree with @yichengt900 that there are some values that should not be changed.  @theresa-morrison, my understanding from our conversation yesterday was that we could set this with an option not to read external values for certain parameters that should not be changed?

Reply:
I obviously didn't read your next comment.....please see additional thoughts below.

Reply:
added do_not_read=true for these variables.